src/vulkan/vulkansetup.cpp

```diff
@@ -267,107 +51,136 @@ bool TempVulkanSetupObject::createVulkanInstance(std::vector<const char*>* desir
+    std::vector<VkPhysicalDevice> returnVec;
+    returnVec.resize(numDevices);
+    if (vkEnumeratePhysicalDevices(instance, &numDevices, returnVec.data()) != VK_SUCCESS) {
         std::cerr << "Could not enumerate physical devices." << std::endl;
-        return false;
+        return std::vector<VkPhysicalDevice>();
     }
 
 #if defined( DW_VERBOSE_LOG_VK )
-    std::cout << "\nFound " << numDevices << " physical device(s):\n";
+    std::cout << "\nFound " << numDevices << " physical device(s):";
     std::cout << std::left << std::setfill(' ');
-    for (const VkPhysicalDevice& device : outAvailableDevices) {
-        prettyPrint(std::cout, device, 30, "\t", "\n");
+    for (const VkPhysicalDevice& physicalDevice : returnVec) {
+        prettyPrint(std::cout, physicalDevice, 30, "\n\t", "");
     }
+    std::cout << std::endl;
 #endif
 
-    return true;
+    return returnVec;
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
-bool TempVulkanSetupObject::getPhysicalDeviceExtensions(const VkPhysicalDevice& device, std::vector<VkExtensionProperties>& outAvailableExtensions)
+std::vector<VkExtensionProperties> GetPhysicalDeviceExtensions(const VkPhysicalDevice& device)
 {
+
     uint32_t numExtensions = 0;
     if (vkEnumerateDeviceExtensionProperties(device, nullptr, &numExtensions, nullptr) != VK_SUCCESS || numExtensions == 0) {
         std::cerr << "Could not get the number of physical device extensions." << std::endl;
-        return false;
+        return std::vector<VkExtensionProperties>();
     }
 
-    outAvailableExtensions.resize(numExtensions);
-    if (vkEnumerateDeviceExtensionProperties(device, nullptr, &numExtensions, outAvailableExtensions.data()) != VK_SUCCESS || numExtensions == 0) {
+    std::vector<VkExtensionProperties> returnVec;
+    returnVec.resize(numExtensions);
+    if (vkEnumerateDeviceExtensionProperties(device, nullptr, &numExtensions, returnVec.data()) != VK_SUCCESS || numExtensions == 0) {
         std::cerr << "Could not enumerate physical device extensions." << std::endl;
-        return false;
+        return std::vector<VkExtensionProperties>();
     }
 
 #if defined( DW_VERBOSE_LOG_VK )
     std::cout << std::left << std::setfill(' ');
-    std::cout << "\nThere are " << numExtensions << " extension properties for \"" << device << "\"\n";
-    for (const VkExtensionProperties& extensionProperties : outAvailableExtensions) {
-        prettyPrint(std::cout, extensionProperties, 40, "\t", "\n");
+    std::cout << "\nThere are " << numExtensions << " extension properties for \"" << device << "\"";
+    for (const VkExtensionProperties& extensionProperties : returnVec) {
+        prettyPrint(std::cout, extensionProperties, 40, "\n\t", "");
     }
+    std::cout << std::endl;
 #endif
 
-    return true;
+    return returnVec;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
```

Comment:
Remove `////////////` stuff? If it's still not temp work?

Reply:
That's actually something I've seen at work.. they separate functions like that. I like it because if I scroll through a file quickly, it's easier to see where a new function starts

Reply:
Okay so in that case we should use `//////////////` between all functions throughout the whole project? 😕

Reply:
Well.. I'm not saying it's a must have.. but I do think it adds to clarity and readability.
If you use the "minimap-scrollbar" (or whatever you want to call it), you see those green lines of "/////" so you can quickly navigate that way too.
![image](https://user-images.githubusercontent.com/7249663/48552269-5a474400-e8d8-11e8-9418-a1ec650bb2ac.png)

Reply:
But if you don't dislike it, I'd say: yeah, add them to the entire project.. but we don't have to be anal about it :P

Reply:
The scrollbar is nice, but I don't see how it provides any value to my workflow at least :P We can have them as optional I guess, but definitely not throughout the whole project, that would be super cumbersome. I'm just thinking what our future style guide should say about these

Reply:
In general I think any deviations from a style guide, that provides clarity/readability is welcome.
They are guides - not rules